### PR TITLE
Enable new CI pipeline for standard & distributed tests

### PR DESCRIPTION
**Pull request by malachi-constant**
### Feature
- Enable new CI pipeline for standard & distributed tests
- Registering & adding marker for distributed tests.
- Adding ray runtime environment variable `__MODIN_AUTOIMPORT_PANDAS__`

By submitting this pull request, I confirm that my contribution is made under the terms of the Apache 2.0 license.

### CODE REVIEW

**Comment by malachi-constant** [comment]
### AWS CodeBuild CI Report

* CodeBuild project: GitHubLoadTests5656BB24-s6u9F3qN9oFy
* Commit ID: 26d6c44bd7d8608869292eef7595708e6458eb77
* Result: SUCCEEDED
* [Build Logs](https://pciyom2pbh.execute-api.us-east-1.amazonaws.com/Prod/buildlogs?key=4d21f7d8-92c5-443f-a8fa-f94203a3dcfa%2Fbuild.log) (available for 30 days)

*Powered by [github-codebuild-logs](https://serverlessrepo.aws.amazon.com/applications/arn:aws:serverlessrepo:us-east-1:277187709615:applications~github-codebuild-logs), available on the [AWS Serverless Application Repository](https://aws.amazon.com/serverless/serverlessrepo/)*

**Comment by malachi-constant** [comment]
### AWS CodeBuild CI Report

* CodeBuild project: GitHubLoadTests5656BB24-s6u9F3qN9oFy
* Commit ID: 155d90771146509b985e5152f24eb385cf350d34
* Result: SUCCEEDED
* [Build Logs](https://pciyom2pbh.execute-api.us-east-1.amazonaws.com/Prod/buildlogs?key=01b5007a-c0f1-4f68-a0dc-af117f8947b4%2Fbuild.log) (available for 30 days)

*Powered by [github-codebuild-logs](https://serverlessrepo.aws.amazon.com/applications/arn:aws:serverlessrepo:us-east-1:277187709615:applications~github-codebuild-logs), available on the [AWS Serverless Application Repository](https://aws.amazon.com/serverless/serverlessrepo/)*

File: awswrangler/distributed/_distributed.py [hunk 1]
```diff
@@ -115,7 +119,11 @@ def initialize_ray(
                 "object_store_memory": object_store_memory,
                 "_redis_password": redis_password,
                 "_memory": object_store_memory,
+                "runtime_env": {
```
Comment (malachi-constant) [inline awswrangler/distributed/_distributed.py:122]: @jaidisido Simply setting the env var at the top-level in CodeBuild doesn't work. Suspicion is the `runtime_env` variables don't get passed down from env vars in the workspace.

**Comment by malachi-constant** [comment]
### AWS CodeBuild CI Report

* CodeBuild project: GitHubLoadTests5656BB24-s6u9F3qN9oFy
* Commit ID: 57982460a7b7129a850d976a933e138b9edbc205
* Result: SUCCEEDED
* [Build Logs](https://pciyom2pbh.execute-api.us-east-1.amazonaws.com/Prod/buildlogs?key=ab6adae0-d2c3-40e4-8b9f-221fdef4447a%2Fbuild.log) (available for 30 days)

*Powered by [github-codebuild-logs](https://serverlessrepo.aws.amazon.com/applications/arn:aws:serverlessrepo:us-east-1:277187709615:applications~github-codebuild-logs), available on the [AWS Serverless Application Repository](https://aws.amazon.com/serverless/serverlessrepo/)*

**Comment by malachi-constant** [comment]
### AWS CodeBuild CI Report

* CodeBuild project: GitHubLoadTests5656BB24-s6u9F3qN9oFy
* Commit ID: cd3d6155900b762aa522b6dd95735d50f1ceef57
* Result: FAILED
* [Build Logs](https://pciyom2pbh.execute-api.us-east-1.amazonaws.com/Prod/buildlogs?key=5fc92e98-da74-400f-a09f-1dbb26221edc%2Fbuild.log) (available for 30 days)

*Powered by [github-codebuild-logs](https://serverlessrepo.aws.amazon.com/applications/arn:aws:serverlessrepo:us-east-1:277187709615:applications~github-codebuild-logs), available on the [AWS Serverless Application Repository](https://aws.amazon.com/serverless/serverlessrepo/)*

**Comment by malachi-constant** [comment]
### AWS CodeBuild CI Report

* CodeBuild project: GitHubLoadTests5656BB24-s6u9F3qN9oFy
* Commit ID: d49bc83f13b2ef8b4885426d40c8e2940646921d
* Result: FAILED
* [Build Logs](https://pciyom2pbh.execute-api.us-east-1.amazonaws.com/Prod/buildlogs?key=908f4f33-0c98-4f48-b7fd-a07ed05b015f%2Fbuild.log) (available for 30 days)

*Powered by [github-codebuild-logs](https://serverlessrepo.aws.amazon.com/applications/arn:aws:serverlessrepo:us-east-1:277187709615:applications~github-codebuild-logs), available on the [AWS Serverless Application Repository](https://aws.amazon.com/serverless/serverlessrepo/)*

File: awswrangler/distributed/_distributed.py [hunk 2]
```diff
@@ -115,6 +119,9 @@ def initialize_ray(
...
```
Comment (malachi-constant) [inline awswrangler/distributed/_distributed.py:122]: Setting this in tox as an environment variable does not get to to `ray_init_kwargs.runtime_env` from what I can tell. So I added this logic to achieve the same.

**Review by cnfait — APPROVED** [review]

**Review by jaidisido — APPROVED** [review]

**Pull request merged**